playwright/src/test/java/com/microsoft/playwright/TestAccessibility.java

```diff
@@ -65,4 +231,255 @@ void keyshortcuts() {
+  }
+
+  @Test
+  void richTextEditableFieldsWithRoleShouldHaveChildren() {
+// TODO:   test.skip(browserName === "webkit", "WebKit rich text accessibility is iffy");
+    page.setContent("<div contenteditable='true' role=\"textbox\">\n" +
+      "  Edit this image: <img src='fakeimage.png' alt='my fake image'>\n" +
+      "</div>");
+    String golden = isFirefox ? "{\n" +
+      "    role: 'textbox',\n" +
+      "    name: '',\n" +
+      "    valueString: 'Edit this image: my fake image',\n" +
+      "    children: [{\n" +
+      "      role: 'text',\n" +
+      "      name: 'my fake image'\n" +
+      "    }]\n" +
+      "  }" : "{\n" +
+      "    role: 'textbox',\n" +
+      "    name: '',\n" +
+      "    valueString: 'Edit this image: ',\n" +
+      "    children: [{\n" +
+      "      role: 'text',\n" +
+      "      name: 'Edit this image:'\n" +
+      "    }, {\n" +
+      "      role: 'img',\n" +
+      "      name: 'my fake image'\n" +
+      "    }]\n" +
+      "  }";
+    AccessibilityNode snapshot = page.accessibility().snapshot();
+    assertNodeEquals(golden, snapshot.children().get(0));
+  }
+
+  // TODO:   suite.skip(browserName === "firefox", "Firefox does not support contenteditable='plaintext-only'");
+// TODO:   suite.skip(browserName === "webkit", "WebKit rich text accessibility is iffy");
+  @Test
+  void plainTextFieldWithRoleShouldNotHaveChildren() {
+    page.setContent("<div contenteditable='plaintext-only' role='textbox'>Edit this image:<img src='fakeimage.png' alt='my fake image'></div>");
+    AccessibilityNode snapshot = page.accessibility().snapshot();
+    assertNodeEquals("{\n" +
+      "  role: 'textbox',\n" +
+      "  name: '',\n" +
+      "  valueString: 'Edit this image:'\n" +
+      "}", snapshot.children().get(0));
+  }
+
+  @Test
+  void plainTextFieldWithoutRoleShouldNotHaveContent() {
+    page.setContent("<div contenteditable='plaintext-only'>Edit this image:<img src='fakeimage.png' alt='my fake image'></div>");
+    AccessibilityNode snapshot = page.accessibility().snapshot();
+    assertNodeEquals("{\n" +
+      "  role: 'generic',\n" +
+      "  name: ''\n" +
+      "}", snapshot.children().get(0));
+  }
+
+  @Test
+  void plainTextFieldWithTabindexAndWithoutRoleShouldNotHaveContent() {
+    page.setContent("<div contenteditable='plaintext-only' tabIndex=0>Edit this image:<img src='fakeimage.png' alt='my fake image'></div>");
+    AccessibilityNode snapshot = page.accessibility().snapshot();
+    assertNodeEquals("{\n" +
+      "  role: 'generic',\n" +
+      "  name: ''\n" +
+      "}", snapshot.children().get(0));
+  }
+  @Test
+  void nonEditableTextboxWithRoleAndTabIndexAndLabelShouldNotHaveChildren() {
+    page.setContent("<div role='textbox' tabIndex=0 aria-checked='true' aria-label='my favorite textbox'>\n" +
+      "this is the inner content\n" +
+      "<img alt='yo' src='fakeimg.png'>\n" +
+      "</div>");
+  String golden = isFirefox ? "{\n" +
```

Comment:
Check indent here and in the rest of the tests :)

Reply:
Done!